src/exe.ml

```diff
@@ -67,6 +67,8 @@ module Linkage = struct
     in
     let ext =
       match wanted_mode, m.kind with
+      | Byte   , C             -> ".bc.c"
+      | Native , C             -> raise (Exn.Fatal_error "C file generation only supports bytecode!")
```

Comment:
I see what you are getting at but I'm not sure how to deal with `Dune_file.Execitables.Link_mode.make`.  To create `exe`, `object_` I'd need a location and I'm not sure how to get it.  Just make a dummy Loc.t var?

Reply:
Just noticed that tests are failing on travis.  I can confirm that:
```bash
$ make release
$ make test-all
```
works on my system (Linux kernel 4.16 and OCaml 4.06.1).  By the error message it looks like it can't find libcamlrun and libtermcap.

Reply:
For the location you can use `Loc.none`. It doesn't really matter as we won't report error for modes created with `make`. We only use `make` with constants to create modes that we know are always valid.

Reply:
I've got this compiling (not committed).  To get the test compiling you need to add obviouisly need loc info.  Isn't this change going to require something like (best exe loc) in dune files?

Reply:
no, the location is implicit. It's basically the location of the part of the s-expression being parsed.

Reply:
They pass on my machine as well, I'm not sure what's happening. You can comment the test with a comment pointing to this discussion.